internal/controllers/cluster/cluster_controller_phases_test.go

```diff
@@ -56,13 +57,30 @@ func TestClusterReconcilePhases(t *testing.T) {
 				},
 			},
 		}
+		clusterNoEndpoint := &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster",
+				Namespace: "test-namespace",
+			},
+			Status: clusterv1.ClusterStatus{
+				InfrastructureReady: true,
+			},
+			Spec: clusterv1.ClusterSpec{
+				InfrastructureRef: &corev1.ObjectReference{
+					APIVersion: "infrastructure.cluster.x-k8s.io/v1beta1",
+					Kind:       "GenericInfrastructureMachine",
```

Comment:
Really weird that we use machines here :D (but fine for me it was the same before)

Reply:
agreed, maybe fix later